control/statesp.py

```diff
@@ -640,125 +640,146 @@ def __rdiv__(self, other):
         raise NotImplementedError(
             "StateSpace.__rdiv__ is not implemented yet.")
 
-    def evalfr(self, omega):
-        """Evaluate a SS system's transfer function at a single frequency.
+    def __call__(self, x, squeeze=True):
+        """Evaluate system's transfer function at complex frequency.
 
-        self._evalfr(omega) returns the value of the transfer function matrix
-        with input value s = i * omega.
+        Returns the complex frequency response `sys(x)` where `x` is `s` for
+        continuous-time systems and `z` for discrete-time systems.
 
-        """
-        warn("StateSpace.evalfr(omega) will be deprecated in a future "
-             "release of python-control; use evalfr(sys, omega*1j) instead",
-             PendingDeprecationWarning)
-        return self._evalfr(omega)
-
-    def _evalfr(self, omega):
-        """Evaluate a SS system's transfer function at a single frequency"""
-        # Figure out the point to evaluate the transfer function
-        if isdtime(self, strict=True):
-            s = exp(1.j * omega * self.dt)
-            if omega * self.dt > math.pi:
-                warn("_evalfr: frequency evaluation above Nyquist frequency")
-        else:
-            s = omega * 1.j
+        To evaluate at a frequency omega in radians per second, enter
+        ``x = omega * 1j``, for continuous-time systems, or
+        ``x = exp(1j * omega * dt)`` for discrete-time systems. Or use
+        :meth:`StateSpace.frequency_response`.
 
-        return self.horner(s)
+        Parameters
+        ----------
+        x: complex or complex array_like
+            Complex frequencies
+        squeeze: bool, optional (default=True)
+            If True and `sys` is single input single output (SISO), returns a
+            1D array or scalar depending on the length of `x`.
```

Comment:
```suggestion
        squeeze : bool, optional (default=True)
            If True and `self` is single input single output (SISO), returns a
            1D array rather than a 3D array.
```